midend/replaceSelectRange.h

```diff
@@ -38,12 +38,21 @@ class DoReplaceSelectRange : public Transform {
     // number of new cases generated for ternary operations due to a range
     // Each case is a key set expression.
     const uint MAX_CASES;
+    bool inSelect = false;
+    // Collects select indices which will need to be replaced with bitcast of
+    // the original value to unsigned. This is needed if we encounter a range
+    // over a signed value at the given index.
+    std::set<int> signedIndicesToReplace;
```

Comment:
why not a set of size_t or unsigned values?
I would change this comment to say "an index i is in this set if selectExpression->components[i] needs to be cast from int<N> to bit<N>". This is only needed if there is a label that has in the i-th position a range expression that contains 0 inside the range".

Reply:
(Because old habits die hard.) Changed to size_t. 

As for the comment, sure, except that we need to replace it any time the value is signed, not only when it crosses over zero. Masks are only defined for `bit` values in P4. I changed it.